homeassistant/components/homekit_controller/button.py

```diff
@@ -46,17 +46,42 @@ class HomeKitButtonEntityDescription(ButtonEntityDescription):
 }
 
 
+@dataclass
```

Comment:
I wouldn't bother adding the skaffolding for generic ecobee buttons for one button type. As we are only expecting one button with this glitch. The declarative style is for when we are expecting lots of very similar characteristics with simple mappings to simple HA entity classes. But this is a workaround for a buggy characteristic, and we've only got one button with this particular bug. So making the declarative version is overkill (so far). It would be better to just do a "standard" entity class.

You define things like the device class directly on the class with the `_attr_` prefix. Then don't bother paramterizing `async_press` at all. Something like this:

```python
class EcobeeClearHoldButton(CharacteristicEntity, ButtonEntity):

    _attr_device_class = ButtonDeviceClass.UPDATE
    # ... snip

    def get_characteristic_types(self):
        """Define the homekit characteristics the entity is tracking."""
        return [CharacteristicsTypes.Vendor.ECOBEE_CLEAR_HOLD]

    # ... snip
    async def async_press(self) -> None:
         for val in (False, True):
            await self.async_put_characteristics({CharacteristicsTypes.Vendor.ECOBEE_CLEAR_HOLD: val})
```

You don't need to implement a description, and you don't need to pass it to this class. So you don't need to even subclass __init__.

Then have:

```python
BUTTON_ENTITY_CLASSES: dict[str, ButtonEntity] = {
    CharacteristicsTypes.Vendor.ECOBEE_CLEAR_HOLD: EcobeeClearHoldButton
}
```

We can make it declarative later if we are flooded with devices that recreate this bug.